src/features/island/delivery/components/Orders.tsx

```diff
@@ -0,0 +1,264 @@
+import { useActor } from "@xstate/react";
+import { SUNNYSIDE } from "assets/sunnyside";
+import React, { useContext, useEffect } from "react";
+import classNames from "classnames";
+import Decimal from "decimal.js-light";
+
+import selectBoxBL from "assets/ui/select/selectbox_bl.png";
+import selectBoxBR from "assets/ui/select/selectbox_br.png";
+import selectBoxTL from "assets/ui/select/selectbox_tl.png";
+import selectBoxTR from "assets/ui/select/selectbox_tr.png";
+import sfl from "assets/icons/token_2.png";
+import heartBg from "assets/ui/heart_bg.png";
+
+import { DynamicNFT } from "features/bumpkins/components/DynamicNFT";
+import { Context } from "features/game/GameProvider";
+import { getDeliverySlots } from "features/game/events/landExpansion/deliver";
+import { PIXEL_SCALE } from "features/game/lib/constants";
+import { getKeys } from "features/game/types/craftables";
+import { Order } from "features/game/types/game";
+import { ITEM_DETAILS } from "features/game/types/images";
+import { NPC } from "features/island/bumpkin/components/NPC";
+
+import { NPC_WEARABLES } from "lib/npcs";
+import { secondsToString } from "lib/utils/time";
+import { generateDeliveryMessage } from "../lib/delivery";
+import { RequirementLabel } from "components/ui/RequirementsLabel";
+import { Button } from "components/ui/Button";
+import { OuterPanel } from "components/ui/Panel";
+
+interface Props {
+  selectedId?: string;
+  onSelect: (id?: string) => void;
+}
+export const DeliveryOrders: React.FC<Props> = ({ selectedId, onSelect }) => {
+  const { gameService } = useContext(Context);
+  const [gameState] = useActor(gameService);
+
+  const delivery = gameState.context.state.delivery;
+  const orders = delivery.orders.filter((order) => Date.now() >= order.readyAt);
+
+  let previewOrder = delivery.orders.find((order) => order.id === selectedId);
+
+  if (!previewOrder) {
+    previewOrder = orders[0];
+  }
+
+  useEffect(() => {
+    console.log("Content render");
+  }, []);
+
+  const deliver = () => {
+    gameService.send("order.delivered", { id: previewOrder?.id });
+    onSelect(undefined);
+  };
+
+  const hasRequirements = (order: Order) => {
+    return getKeys(order.items).every((name) => {
+      const count = gameState.context.state.inventory[name] || new Decimal(0);
+      const amount = order.items[name] || new Decimal(0);
+
+      return count.gte(amount);
+    });
+  };
+
+  const select = (id: string) => {
+    onSelect(id);
+  };
+
+  if (orders.length === 0) {
+    return <p>No orders available</p>;
+  }
+
+  const canFulfill = hasRequirements(previewOrder as Order);
```

Comment:
```suggestion
  const canFulfill = !!previewOrder && hasRequirements(previewOrder);
```

**suggestion:** It may be better to show when the next order is upcoming rather than the No orders available screen.

<img width="792" alt="screenshot" src="https://github.com/sunflower-land/sunflower-land/assets/41215134/02df2f50-35c7-490f-b4b1-513d99cf1d60">

Reply:
Good catch. This is a fallback scenario, realistically we should always have an order pending/preparing.

I have improved the UI though so it looks nicer 👍

Reply:
Yup! `orders.length` can be 0 if there are pending orders - as the value of `orders` filters pending further up in the function.